File: examples/python/src/services/user_service.py
from typing import Optional, List
from datetime import datetime
from ..models.user import User


class UserService:
    """Service for managing user operations."""
    
    def __init__(self):
        self._users: List[User] = []
    
    def get_user_by_id(self, user_id: int) -> Optional[User]:
        """Retrieve a user by their ID."""
        for user in self._users:
            if user.id == user_id:
                return user
        return None
    
    def create_user(self, email: str, name: str) -> User:
        """Create a new user."""
        user_id = len(self._users) + 1
        user = User(
            id=user_id,
            email=email,
            name=name,
            created_at=datetime.now()
        )
        self._users.append(user)
        return user

File: examples/nodejs/src/UserService.ts
export interface User {
  id: string;
  email: string;
  name: string;
  createdAt: Date;
}

export class UserService {
  private users: User[] = [];

  getUserById(id: string): User | null {
    return this.users.find(user => user.id === id) ?? null;
  }

  createUser(email: string, name: string): User {
    const user: User = {
      id: String(this.users.length + 1),
      email,
      name,
      createdAt: new Date()
    };
    this.users.push(user);
    return user;
  }
}
